### ISSUE TITLE
Beacon Node exited with code 1 after REST post on /eth/v1/beacon/blocks

### ISSUE THREAD

**Issue by BeroBurny** [status-im/nim-beacon-chain#2603]
Platform details (OS, architecture):  `Ubuntu 20.04.2 LTS` - `x86_64`
Branch/commit used:  `unstable` - `d69e06e519827644097f301283789146135d7e22`

**Describe the bug**
Proposing a block with an external validator over a REST protocol causes a crashing beacon node.

<details>
  <summary>Relevant log from first crash</summary>

> DBG 2021-05-26 06:15:41.005+00:00 Remote peer disconnected                   topics="beacnde" tid=97 file=serverprivate.nim:162 peer=172.28.0.1:53322 reason=DisconnectError
DBG 2021-05-26 06:15:41.006+00:00 Received request                           topics="beacnde" tid=97 file=serverprivate.nim:38 peer=172.28.0.1:53326 meth=GET uri="/eth/v1/validator/blocks/49?randao_reveal=0xa6535acd8ea9d47b32ce9fe73c006fca37e72bc822b201a1e16c1c317d3a62bb28b25085b2f81a1adda7b83905bdacff07cf46d5ee4f590f24b4f9c2b941edba08797cc2f66668bd8462ff65eac895489f550ea462f7f523d3e16eca70f25817&graffiti=0x436861696e477561726469616e00000000000000000000000000000000000000"
DBG 2021-05-26 06:15:41.006+00:00 Serving API request                        topics="beacnde" tid=97 file=serverprivate.nim:66 peer=172.28.0.1:53326 meth=GET uri="/eth/v1/validator/blocks/49?randao_reveal=0xa6535acd8ea9d47b32ce9fe73c006fca37e72bc822b201a1e16c1c317d3a62bb28b25085b2f81a1adda7b83905bdacff07cf46d5ee4f590f24b4f9c2b941edba08797cc2f66/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncloop.nim(340) main
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncloop.nim(333) NimMain
�X9D|U(2030)             main
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncloop.nim(1689) doRunBeaconNode
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1319) start
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1266) run
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(279) poll
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/apps/http/httpserver.nim(612) processLoop
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(101) processRestRequest
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(77) colonanonymous
/home/user/nimbus-eth2/vendor/nim-presto/presto/serverprivate.nim(73) processRestRequest
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(101) restApiEthV1BeaconBlocksPostHandler
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(626) restApiEthV1BeaconBlocksPostHandler
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(337) proposeSignedBlock
/home/user/nimbus-eth2/beacon_chain/consensus_object_pools/block_clearance.nim(337) addRawBlock
/home/user/nimbus-eth2/beacon_chain/consensus_object_pools/block_clearance.nim(246) addRawBlockKnownParent
/home/user/nimbus-eth2/beacon_chain/consensus_object_pools/block_clearance.nim(94) addResolvedBlock
/home/user/nimbus-eth2/beacon_chain/consensus_object_pools/blockchain_dag.nim(195) link
/home/user/nimbus-eth2/vendor/nimbus-build-system/vendor/Nim/lib/system/assertions.nim(29) failedAssertImpl
/home/user/nimbus-eth2/vendor/nimbus-build-system/vendor/Nim/lib/system/assertions.nim(22) raiseAssert
/home/user/nimbus-eth2/vendor/nimbus-build-system/vendor/Nim/lib/system/fatal.nim(49) sysFatal
[[reraised from:
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(340) main
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(333) NimMain
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(2030) main
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1689) doRunBeaconNode
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1319) start
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1266) run
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(279) poll
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/apps/http/httpserver.nim(612) processLoop
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(101) processRestRequest
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(77) colonanonymous
/home/user/nimbus-eth2/vendor/nim-presto/presto/serverprivate.nim(73) processRestRequest
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(101) restApiEthV1BeaconBlocksPostHandler
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(96) colonanonymous
]]
[[reraised from:
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(340) main
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(333) NimMain
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(2030) main
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1689) doRunBeaconNode
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(1319) start
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(1266) run
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(279) poll
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/apps/http/httpserver.nim(612) processLoop
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(101) processRestRequest
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(77) colonanonymous
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(252) processRestRequest
]]
[[reraised from:
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(340) main
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(333) NimMain
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(2030) main
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1689) doRunBeaconNode
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1319) start
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1266) run
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncloop.nim(279) poll
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/apps/http/httpserver.nim(612) processLoop
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(101) processRestRequest
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(96) colonanonymous
]]
[[reraised from:
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(340) main
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(333) NimMain
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(2030) main
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1689) doRunBeaconNode
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1319) start
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1266) run
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncloop.nim(279) poll
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(74) colonanonymous
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(252) processLoop
]]
[[reraised from:
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(340) main
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/client.nim(333) NimMain
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(2030) main
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1689) doRunBeaconNode
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1319) start
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1266) run
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncloop.nim(279) poll
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncmacro2.nim(96) colonanonymous
]]
Error: unhandled exception: /home/user/nimbus-eth2/beacon_chain/consensus_object_pools/blockchain_dag.nim(195, 12) `(not (parent.root == Eth2Digest() or child.root == Eth2Digest()))` blocks missing root! [AssertionError]
668bd8462ff65eac895489f550ea462f7f523d3e16eca70f25817&graffiti=0x436861696e477561726469616e00000000000000000000000000000000000000" path_params="[(slot, 49)]" query_params="[(randao_reveal, 0xa6535acd8ea9d47b32ce9fe73c006fca37e72bc822b201a1e16c1c317d3a62bb28b25085b2f81a1adda7b83905bdacff07cf46d5ee4f590f24b4f9c2b941edba08797cc2f66668bd8462ff65eac895489f550ea462f7f523d3e16eca70f25817), (graffiti, 0x436861696e477561726469616e00000000000000000000000000000000000000)]" content_body=(None)
DBG 2021-05-26 06:15:41.007+00:00 Packed attestations for block              topics="attpool" tid=97 file=attestation_pool.nim:523 newBlockSlot=49 packingTime=161us729ns totalCandidates=1 attestations=1
DBG 2021-05-26 06:15:41.007+00:00 Received response from handler             topics="beacnde" tid=97 file=serverprivate.nim:100 status="200 OK" meth=GET peer=172.28.0.1:53326 uri="/eth/v1/validator/blocks/49?randao_reveal=0xa6535acd8ea9d47b32ce9fe73c006fca37e72bc822b201a1e16c1c317d3a62bb28b25085b2f81a1adda7b83905bdacff07cf46d5ee4f590f24b4f9c2b941edba08797cc2f66668bd8462ff65eac895489f550ea462f7f523d3e16eca70f25817&graffiti=0x436861696e477561726469616e00000000000000000000000000000000000000" content_type=application/json content_size=1369
DBG 2021-05-26 06:15:41.012+00:00 Remote peer disconnected                   topics="beacnde" tid=97 file=serverprivate.nim:162 peer=172.28.0.1:53326 reason=DisconnectError
DBG 2021-05-26 06:15:41.013+00:00 Received request                           topics="beacnde" tid=97 file=serverprivate.nim:38 peer=172.28.0.1:53330 meth=POST uri=/eth/v1/beacon/blocks
DBG 2021-05-26 06:15:41.013+00:00 Serving API request                        topics="beacnde" tid=97 file=serverprivate.nim:66 peer=172.28.0.1:53330 meth=POST uri=/eth/v1/beacon/blocks path_params=[] query_params=[] content_body="(application/json, 1581 bytes)"

</details>

**To Reproduce**
1. run docker image with local testnet settings
2. run external validator client connected to beacon node over a REST
3. wait for a validator to become the proposer

But to have indicial reproduction to reproduce is to clone `ChainGuardian` repo on branch `beroburny/nimbus`
=> https://github.com/NodeFactoryIo/ChainGuardian/tree/beroburny/nimbus commit `c9bffdac036760c2d99ae91c0fd425c2c484de39`
(requirements node `12.13.0` and `yarn`)
1. `yarn install`
2. `docker-compose -f nimbus-testnet.yml up`
3. `yarn run integration:nimbus`

[docker compose file](https://github.com/NodeFactoryIo/ChainGuardian/blob/beroburny/nimbus/nimbus-testnet.yml)
[tesnet nimbus settings](https://github.com/NodeFactoryIo/ChainGuardian/tree/beroburny/nimbus/testnet/nimbus)

**LOG'S**
[log0.txt from first crash](https://github.com/status-im/nimbus-eth2/files/6544603/log0.txt)
[log0.txt from second crash](https://github.com/status-im/nimbus-eth2/files/6544684/log0.txt)

**Comment by BeroBurny** [comment]
post **JSON example**
```json
{
  "message":{
    "slot":"49",
    "proposer_index":"29",
    "parent_root":"0x1bd61e54d5b721cb0002eb83998b12dc27629e51c81b194164e222043e2bf327",
    "state_root":"0xa03a448bd0ee36d40acc57fc173bc22eac80e7dec7c2979b84ca1f55633a7b99",
    "body":{
      "randao_reveal":"0xa6535acd8ea9d47b32ce9fe73c006fca37e72bc822b201a1e16c1c317d3a62bb28b25085b2f81a1adda7b83905bdacff07cf46d5ee4f590f24b4f9c2b941edba08797cc2f66668bd8462ff65eac895489f550ea462f7f523d3e16eca70f25817",
      "eth1_data":{
        "deposit_root":"0x2c11be57935bf837c7a7f512e26a3479cce591a078944e511e598da35423b76f",
        "deposit_count":"32",
        "block_hash":"0x4242424242424242424242424242424242424242424242424242424242424242"
      },
      "graffiti":"0x436861696e477561726469616e00000000000000000000000000000000000000",
      "proposer_slashings":[
        
      ],
      "attester_slashings":[
        
      ],
      "attestations":[
        {
          "aggregation_bits":"0x03",
          "data":{
            "slot":"48",
            "index":"0",
            "beacon_block_root":"0x1bd61e54d5b721cb0002eb83998b12dc27629e51c81b194164e222043e2bf327",
            "source":{
              "epoch":"0",
              "root":"0x0000000000000000000000000000000000000000000000000000000000000000"
            },
            "target":{
              "epoch":"1",
              "root":"0x3892eb7751b2a54c44f24c8fa2bb129c394c0c517f3fdf250d0cc725b219ee4b"
            }
          },
          "signature":"0xa96b0f8625cce1bfe95ebde90197b9e546f9dc70bf0432ffc84ef12b985783d56bbf0e60ebc2d00e86105599f02edfc000e5ae72c5910560c3e8c6935d1a72e895b5894ce7de81e73a00a981c6e6f1fac877006b493d0a572b64105faa89d38a"
        }
      ],
      "deposits":[
        
      ],
      "voluntary_exits":[
        
      ]
    }
  },
  "signature":"0x83376c9996aeb1b23c9269fd53ecad237ce25966fd86ea9a4d76e54e8f4d55ed47d6f5542d40285729ff99dd4acdde1709080928f3ac3c1c6d708b0c017a50d793bc0a2e65b769373f6360ac73d782bca3b69e12e91a56c2b407ba7fea2fe38d"
}
```

[log0.txt](https://github.com/status-im/nimbus-eth2/files/6552885/log0.txt)

**case 2 with missed epoch `1`**
```json
{
  "message":{
    "slot":"114",
    "proposer_index":"29",
    "parent_root":"0x77a01762e0845b5c7b3517b74204faa963a56e69930f61f15587b2ce5fb9feed",
    "state_root":"0x0c7a482b65eb129fa59a59a2ef3cdd5649627e5100e5fc1a145173d4476daed8",
    "body":{
      "randao_reveal":"0x80ba78f9b0972da5581ae04b6807577b74f74973472b25ee8cd1c51276b92b62bec81fb44d0a55b615e2eec5fe0281f50decd49860f38e83cb968e76f79922e6ed7fae46e99e36132460e34976aedd113d34b71bb52a6d512af3e243820a4584",
      "eth1_data":{
        "deposit_root":"0x2c11be57935bf837c7a7f512e26a3479cce591a078944e511e598da35423b76f",
        "deposit_count":"32",
        "block_hash":"0x4242424242424242424242424242424242424242424242424242424242424242"
      },
      "graffiti":"0x436861696e477561726469616e00000000000000000000000000000000000000",
      "proposer_slashings":[
        
      ],
      "attester_slashings":[
        
      ],
      "attestations":[
        {
          "aggregation_bits":"0x03",
          "data":{
            "slot":"113",
            "index":"0",
            "beacon_block_root":"0x77a01762e0845b5c7b3517b74204faa963a56e69930f61f15587b2ce5fb9feed",
            "source":{
              "epoch":"2",
              "root":"0x589152aa0ab3575da0b08742a6fbb4cf89e6bbbc71c538be50c80f5277c1b7b6"
            },
            "target":{
              "epoch":"3",
              "root":"0x21d4358f6e4bae7954f6ba166c9a90d332e0d3f6c0f9cb909eda0b16a46eb4e5"
            }
          },
          "signature":"0x8f39985c3aa7b1338d9496f27cd1a304c3a2487622dd0269811ab862f708c8aa3124f66bc99996c0f9a5f626f1f37589006ba1cb53a406f0766602014e0210b2270275e95164f5f22365ea7c771e8544331a38f15b929124932f5b882290ce99"
        }
      ],
      "deposits":[
        
      ],
      "voluntary_exits":[
        
      ]
    }
  },
  "signature":"0x9189c998d5b4abc9821dfafe14a530c613886bd5d61b02e7d98b21104bd6791baf2fc16d910a7897dd505fe0fe852c320ceebf2be35760e8cfd89db6002479fdea71dc6f2698a26a52303a97ef02fc6e98aef57e7d0c1c63701aca5ca905c690"
}
```

[log0.txt](https://github.com/status-im/nimbus-eth2/files/6553119/log0.txt)

**Comment by cheatfate** [comment]
Was fixed by #2605
